lib/ansible/modules/find.py

```diff
@@ -431,15 +442,16 @@ def main():
             module.fail_json(size=params['size'], msg="failed to process size")
 
     now = time.time()
-    msg = ''
+    msg = 'All paths examined'
     looked = 0
+    has_warnings = False
     for npath in params['paths']:
         npath = os.path.expanduser(os.path.expandvars(npath))
         try:
             if not os.path.isdir(npath):
                 raise Exception("'%s' is not a directory" % to_native(npath))
 
-            for root, dirs, files in os.walk(npath, followlinks=params['follow']):
+            for root, dirs, files in os.walk(npath, onerror=handle_walk_errors, followlinks=params['follow']):
```

Comment:
I think this'd read better with a bit of semantic naming:
```suggestion
            for root, dirs, files in os.walk(npath, onerror=raise_walk_error, followlinks=params['follow']):
```